src/platforms/launchpad-target/platform.h

```diff
@@ -0,0 +1,155 @@
+#define SWDIO_PIN	TMS_PIN
+
+#define SWCLK_PORT	TCK_PORT
+#define SWCLK_PIN	TCK_PIN
+
+#define SRST_PORT	GPIOA_BASE
+#define SRST_PIN	GPIO6
+
+#define LED_PORT	GPIOF_BASE
+#define LED_ERROR	GPIO1
+#define LED_IDLE	GPIO2
+#define LED_RUN		GPIO3
+
+#define TMS_SET_MODE()	{								\
+	gpio_mode_setup(TMS_PORT, GPIO_MODE_OUTPUT, GPIO_PUPD_NONE, TMS_PIN);		\
+	gpio_set_output_config(TMS_PORT, GPIO_OTYPE_PP, GPIO_DRIVE_2MA, TMS_PIN);	\
+}
+
+#define SWDIO_MODE_FLOAT() {								\
+	gpio_mode_setup(SWDIO_PORT, GPIO_MODE_INPUT, GPIO_PUPD_NONE, SWDIO_PIN);	\
+}
+
+#define SWDIO_MODE_DRIVE() {									\
+	gpio_mode_setup(SWDIO_PORT, GPIO_MODE_OUTPUT, GPIO_PUPD_NONE, SWDIO_PIN);		\
+	gpio_set_output_config(SWDIO_PORT, GPIO_OTYPE_PP, GPIO_DRIVE_2MA, SWDIO_PIN);		\
+}
+
+extern const usbd_driver lm4f_usb_driver;
+#define USB_DRIVER	lm4f_usb_driver
+#define USB_IRQ		NVIC_USB0_IRQ
+#define USB_ISR		usb0_isr
+
+#define IRQ_PRI_USB	(2 << 4)
+
+#define USBUART		UART5
+#define USBUART_CLK	RCC_UART5
+#define USBUART_IRQ	NVIC_UART5_IRQ
+#define USBUART_ISR	uart5_isr
+#define UART_PIN_SETUP() do {								\
+	periph_clock_enable(RCC_GPIOE);							\
+	__asm__("nop"); __asm__("nop"); __asm__("nop");					\
+	gpio_set_af(GPIOE_BASE, 0x1, GPIO4 | GPIO5);				\
+	gpio_mode_setup(GPIOE_BASE, GPIO_MODE_INPUT, GPIO_PUPD_NONE, GPIO4);	\
+	gpio_mode_setup(GPIOE_BASE, GPIO_MODE_OUTPUT, GPIO_PUPD_NONE, GPIO5);	\
+	} while (0)
+
+#define TRACEUART	UART7
+#define TRACEUART_CLK	RCC_UART7
+#define TRACEUART_PORT_CLK	RCC_GPIOE
+#define TRACEUART_IRQ	NVIC_UART7_IRQ
+#define TRACEUART_ISR	uart7_isr
+
+/* Use newlib provided integer only stdio functions */
+#define sscanf siscanf
+#define sprintf siprintf
+#define vasprintf vasiprintf
+#define snprintf sniprintf
+
+#ifdef ENABLE_DEBUG
+
+#define DEBUG printf
+
+#define DEBUGUART		UART0
+#define DEBUGUART_CLK	RCC_UART0
+#define DEBUGUART_IRQ	NVIC_UART0_IRQ
+#define DEBUGUART_ISR	uart0_isr
+#define DEBUGUART_PIN_SETUP() do {								\
+	periph_clock_enable(RCC_GPIOA);							\
+	__asm__("nop"); __asm__("nop"); __asm__("nop");					\
+	gpio_set_af(GPIOA_BASE, 0x1, GPIO0 | GPIO1);				\
+	gpio_mode_setup(GPIOA_BASE, GPIO_MODE_OUTPUT, GPIO_PUPD_NONE, GPIO0);	\
+	gpio_mode_setup(GPIOA_BASE, GPIO_MODE_INPUT, GPIO_PUPD_NONE, GPIO1);	\
+	} while (0)
+
+void debuguart_init(void);
+void debuguart_test(void);
+
+#else
+#define DEBUG(...)
+#endif
+
+#define SET_RUN_STATE(state)	{gpio_set_val(LED_PORT, LED_RUN, state);running_status = (state);}
+#define SET_IDLE_STATE(state)	{gpio_set_val(LED_PORT, LED_IDLE, state);}
+#define SET_ERROR_STATE(state)	{gpio_set_val(LED_PORT, LED_ERROR, state);}
+
+#define PLATFORM_HAS_TRACESWO
+
+inline static void gpio_set_val(uint32_t port, uint8_t pin, uint8_t val) {
+	gpio_write(port, pin, val == 0 ? 0 : 0xff);
+}
+
+inline static uint8_t gpio_get(uint32_t port, uint8_t pin) {
+	return !(gpio_read(port, pin) == 0);
+}
+
+#define disconnect_usb() do { usbd_disconnect(usbdev,1); nvic_disable_irq(USB_IRQ);} while(0)
```

Comment:
Consider writing this as a function. Functions are clearer to read and do more checking.

Reply:
Function removed, as it was not used (copy-paste leftover)